src/bp-members/classes/class-bp-signup.php

```diff
@@ -1084,4 +980,161 @@ public static function delete( $signup_ids = array() ) {
+				}
+			} else {
+
+				// Check user status before sending email.
+				$user_id = email_exists( $signup->user_email );
+
+				if ( ! empty( $user_id ) && 2 !== self::check_user_status( $user_id ) ) {
+
+					// Status is not 2, so user's account has been activated.
+					$result['errors'][ $signup->signup_id ] = array( $signup->user_login, esc_html__( 'the sign-up has already been activated.', 'buddypress' ) );
+
+					// Repair signups table.
+					self::validate( $signup->activation_key );
+
+					continue;
+
+					// Send the validation email.
+				} else {
+					$salutation = $signup->user_login;
+					if ( bp_is_active( 'xprofile' ) && isset( $meta[ 'field_' . bp_xprofile_fullname_field_id() ] ) ) {
+						$salutation = $meta[ 'field_' . bp_xprofile_fullname_field_id() ];
+					}
+
+					bp_core_signup_send_validation_email( false, $signup->user_email, $signup->activation_key, $salutation );
+				}
+			}
+
+			// Update metas.
+			$result['resent'][] = self::update(
+				array(
+					'signup_id' => $signup->signup_id,
+					'meta'      => $meta,
+				)
+			);
+		}
+
+		/**
+		 * Fires after activation emails are resent.
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param array $signup_ids Array of IDs to resend activation emails to.
+		 * @param array $result     Updated metadata related to activation emails.
+		 */
+		do_action( 'bp_core_signup_after_resend', $signup_ids, $result );
+
+		/**
+		 * Filters the result of the metadata for signup activation email resends.
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param array $result Updated metadata related to activation emails.
+		 */
+		return apply_filters( 'bp_core_signup_resend', $result );
+	}
+
+	/**
+	 * Check if an activation email can be resent.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param BP_Signup $signup The signup object.
+	 * @return bool
+	 */
+	public static function resend_activation( $signup ) {
+
+		// Bail if the signup is not a BP_Signup object.
+		if ( ! $signup instanceof BP_Signup ) {
+			return false;
+		}
+
+		// Allow the activation email to be sent if not already.
+		if ( ! $signup->recently_sent || ! $signup->count_sent ) {
+			return true;
+		}
+
+		$sent_at = mysql2date( 'U', $signup->date_sent );
+		$now     = time();
+		$diff    = $now - $sent_at;
+
+		/**
+		 * Filters the lock time for the resend activation.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param int $lock_time The lock time for the resend activation.
+		 */
+		$lock_time = apply_filters( 'bp_core_signup_resend_activation_lock_time', HOUR_IN_SECONDS );
+
+		// If the activation email was sent less than the lock time ago.
+		$retval = false === ( $diff < $lock_time );
+
+		/**
+		 * Filters whether an activation email can be resent.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool      $retval Whether an activation email can be resent.
+		 * @param BP_Signup $signup The signup object.
+		 */
+		return apply_filters( 'bp_core_signup_resend_activation', $retval, $signup );
```

Comment:
I chose not to name this `bp_core_signup_resend_activation_email`, because a site could be using a different type of notification.